programs/scope/src/states.rs

```diff
@@ -115,26 +117,31 @@ pub struct OraclePrices {
     pub prices: [DatedPrice; MAX_ENTRIES],
 }
 
+// Maximum number of sources per token
+pub const MAX_SOURCES: usize = 3;
+
 static_assertions::const_assert_eq!(ORACLE_MAPPING_SIZE, std::mem::size_of::<OracleMappings>());
 static_assertions::const_assert_eq!(0, std::mem::size_of::<OracleMappings>() % 8);
+
 #[account(zero_copy)]
 #[derive(Debug, AnchorDeserialize)]
 pub struct OracleMappings {
-    pub price_info_accounts: [Pubkey; MAX_ENTRIES],
-    pub price_types: [u8; MAX_ENTRIES],
-    pub twap_source: [u16; MAX_ENTRIES], // meaningful only if type == TWAP; the index of where we find the TWAP
-    pub twap_enabled: [u8; MAX_ENTRIES], // true or false
+    // Each token can have up to MAX_SOURCES sources
+    pub price_info_accounts: [[Pubkey; MAX_SOURCES]; MAX_ENTRIES],
+    pub price_types: [[u8; MAX_SOURCES]; MAX_ENTRIES],
+    pub twap_source: [[u16; MAX_SOURCES]; MAX_ENTRIES], // meaningful only if type == TWAP; the index of where we find the TWAP
+    pub twap_enabled: [[u8; MAX_SOURCES]; MAX_ENTRIES], // true or false
     pub ref_price: [u16; MAX_ENTRIES], // reference price against which we check confidence within 5%
-    pub generic: [[u8; 20]; MAX_ENTRIES], // generic data parsed depending on oracle type
+    pub generic: [[[u8; 20]; MAX_SOURCES]; MAX_ENTRIES], // generic data parsed depending on oracle type
 }
```

Comment:
This modifies the states in a way that makes it incompatible with the current version. This program is deployed and in use, and we can't introduce this change in the way you suggest.

Reply:
@oeble Thank you for reviewing my PR and for your feedback. I understand the concerns regarding state incompatibility with the current deployed version of the program.
I'm interested in exploring ways to implement support for multiple oracle sources per token in a backward-compatible manner. I also recognize that a more sophisticated approach to price aggregation is desired.
Could you provide more details on how you envision the rules for electing or computing the final price from multiple sources? I'd be happy to collaborate and adjust my implementation to align with the project's roadmap and requirements. I am reviewing other repositories as well, if you need any part that need to be reviewed most, please let me know. Thanks

Reply:
Closing this PR, I will move to other solution